llm/ggml.go

```diff
@@ -527,3 +529,40 @@ func (llm GGML) GraphSize(context, batch uint64) (kv, partialOffload, fullOffloa
 
 	return
 }
+
+// SupportsKVCacheType checks if the requested cache type is supported
+func (ggml GGML) SupportsKVCacheType(cacheType string) bool {
+	validKVCacheTypes := []string{"f16", "q8_0", "q4_0"}
+	return slices.Contains(validKVCacheTypes, cacheType)
```

Comment:
This triggers a warning if flash attention is enabled but KV cache type is left as the default because cacheType is the empty string. One option is to normalize the value that comes out of the environment variable so we don't have to check for default empty values everywhere.

There are several places where we check for both f16 and fp16 but here we will only accept f16. I think that's fine - there's no real reason to accept both but we might as well remove fp16 as an option from other places since we will never hit it.

Reply:
I've gone through and:

- Removed references to fp16.
- Defaulted to f16, removing the need for the "" check.
- Added a warning if you're requested a quantised k/v cache but have FA disabled.

Thanks for the review @jessegross!